src/index.ts

```diff
@@ -104,17 +146,33 @@ export class SgidClient {
    * @param code The authorization code received from the authorization server
    * @param nonce Specify null if no nonce
    * @param redirectUri The redirect URI used in the authorization request, defaults to the one registered with the client
+   * @param codeVerifier The code verifier that was used to generate the code challenge that was passed in `authorizationUrl`
    * @returns The sub of the user and access token
    */
-  async callback(
-    code: string,
-    nonce: string | null = null,
+  async callback({
+    code,
+    nonce = null,
     redirectUri = this.getFirstRedirectUri(),
-  ): Promise<{ sub: string; accessToken: string }> {
+    codeVerifier,
+  }: CallbackParams): Promise<{ sub: string; accessToken: string }> {
+    if (this.apiVersion !== 2) {
+      // eslint-disable-next-line typesafe/no-throw-sync-func
+      throw new Error(
+        "Code verifier must be provided in 'callback' when using apiVersion 2",
+      )
+    }
+
+    if (codeVerifier === undefined) {
+      // eslint-disable-next-line typesafe/no-throw-sync-func
```

Comment:
Question: I notice we disable this specific lint rule every time we throw an Error in the codebase - should we just refactor and disable this rule in config instead?

Reply:
yeah i think i'll do that since this has been occurring multiple times (unless someone has objections)

Reply:
after some discussion we've decided that this ESLint rule, while it does have its use, is quite opinionated and not necessarily considered a standard practice (many Python and Node libraries throw errors in synchronous functions)

additionally, as our other codebases are not adhering to this specific ESLint rule, we've decided not to adhere to it as part of an effort to standardize our coding practices